### PR TITLE
Feat/fcm initial

### PR DESCRIPTION
**Pull request by khmak3**
Please download google-services.json from below url and put into android/app/ for all firebase keys
https://console.firebase.google.com/u/0/project/vote4hk/settings/general/android:com.example.vote4hk_mobile

Please goto below URL to adjust any API restriction. Firebase done a bad job for the auto adjustment
https://console.cloud.google.com/apis/credentials?project=vote4hk

### CODE REVIEW

**Comment by khmak3** [comment]
You can go to https://console.firebase.google.com/u/0/project/vote4hk/notification to use a sample cloud message to send a notification to client

File: lib/services/user_service.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+class UserService {
+  SharedPreferences _sharedPreferences;
+  final FirebaseMessaging _firebaseMessaging = FirebaseMessaging();
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+  // constructor
+  UserService() {
+    SharedPreferences.getInstance().then((value) {
+      _sharedPreferences = value;
+      String _userID = _sharedPreferences.getString('userID');
```
Comment (nandiheath) [inline lib/services/user_service.dart:21]: only private variable should prefix underline

File: lib/services/user_service.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:uuid/uuid.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import '../models/user.dart';
+
+User currentUser;
+UserService userService;
+
+class UserService {
```
Comment (nandiheath) [inline lib/services/user_service.dart:13]: i would prefer abstracting this class. e.g. `FirebaseUserService` implements `UserService` and we have kind of factory method for easily switch from firebase to other datastore.

File: lib/services/user_service.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  UserService() {
+    SharedPreferences.getInstance().then((value) {
+      _sharedPreferences = value;
+      String _userID = _sharedPreferences.getString('userID');
+      checkIsSignedIn(_userID);
+    });
+  }
+
+  //Checks if the user has signed in
+  Future<void> checkIsSignedIn(String _userID) async {
```
Comment (nandiheath) [inline lib/services/user_service.dart:27]: i suggest params should have no `_` prefix.

File: lib/services/user_service.dart [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    String userID = _userID;
```
Comment (nandiheath) [inline lib/services/user_service.dart:28]: this is not necessary?

**Review by nandiheath — CHANGES_REQUESTED** [review]

**Pull request closed without merging**